aten/src/ATen/native/LegacyDefinitions.cpp

```diff
@@ -51,15 +51,33 @@ Tensor& zero_(Tensor& self) {
 }
 
 Tensor & masked_fill_(Tensor& self, const Tensor & mask, Scalar value) {
-  return at::legacy::th::_th_masked_fill_(self, mask, value);
+  // As we dispatch on self and TH is type-chcked, we need different definitions.
+  // This can be fixed by moving to ATen.
+  if (mask.dtype() == at::ScalarType::Byte) {
+    return at::legacy::th::_th_masked_fill_(self, mask, value);
```

Comment:
in a future PR, it might make sense to call the bool one masked_fill_ and the other one masked_fill_byte, since that is the end state, but not necessary for now.